tests/system/conftest.py

```diff
@@ -52,12 +52,25 @@ def ds_client(to_delete, deleted_keys):
         client.delete_multi(to_delete)
         deleted_keys.update(to_delete)
 
-    results = [
-        entity
-        for entity in all_entities(client)
-        if entity.key not in deleted_keys
-    ]
-    assert not results
+    while True:
```

Comment:
Do we want to limit this with some sort of maximum timeout, in case there is a test failure and the result is an infinite loop instead of an exception?

Reply:
It couldn't hurt.

Reply:
@andrewsg eee9942